docs/source/run_model_in_simulation.rst

```diff
@@ -0,0 +1,204 @@
+--------------------------------------------------
+
+gem5-Aladdin, which provides the simulation capabilities of estimating
+performance/power/area for pre-RTL kernels via various SoCs, is a trace-driven
+simulator. Thus, in SMAUG, in order to run a model in gem5-Aladdin, we need to
+generate a dynamic trace for the kernels to be simulated as hardware blocks.
+
+First, we need to build the SMAUG tracer, which is an instrumented binary that
+will be executed to generate the dynamic trace::
+
+    make tracer -j4
+
+After running the above command, we will get a binary under :code:`build/bin/`
+named :code:`smaug-instrumented`. Then we will run the model created in
+:ref:`label_build_python_model` using the following command::
+
+    build/bin/smaug-instrumented my_model_topo.pbtxt my_model_params.pb
+
+This generates a trace file named :code:`dynamic_trace_acc0.gz`.
+
+Create gem5-Aladdin configuraion files
+---------------------------------------
+
+gem5-Aladdin also requires two configuration files for running simulations. One
+file specifies SoC-level parameters such as each accelerator's ID, cache or DMA
+configurations, location of the corresponding dynamic trace file, and etc.  An
+example can found be in :code:`experiments/sims/smv/minerva/gem5.cfg`. The
+other file gives the parameters to be used for implementing the accelerator,
+such as parameters for applying loop unrolling and scratchpad partition. An
+example configuration file of implementing the hardware blocks in our `SMV`
+backend can be found in :code:`experiments/sims/smv/smv-accel.cfg`. We defer to
+the `gem5-Aladdin GitHub repo <https://github.com/harvard-acc/gem5-aladdin>`_
+and `gem5-Aladdin tutorial
+<http://accelerator.eecs.harvard.edu/micro16tutorial/slides/micro2016-tutorial-gem5-aladdin.pptx>`_
+for more details of writing these configuration files.
+
+Run the first simulation
+------------------------
+
+Assuming we use configuration files in
+:code:`experiments/sims/smv/tests/minerva/gem5.cfg`
+and :code:`experiments/sims/smv/smv-accel.cfg`, let's create a folder for our
+first simulation. It contains these files::
+
+    my_model_topo.pbtxt my_model_params.pb dynamic_trace_acc0.gz gem5.cfg smv-accel.cfg
+
+Now we are ready to launch the simulation::
+
+    <path-to-gem5-aladdin>/build/X86/gem5.opt \
+      --debug-flags=Aladdin,HybridDatapath \
+      --outdir=outputs \
+      <path-to-gem5-aladdin>/configs/aladdin/aladdin_se.py \
+      --num-cpus=1 \
+      --mem-size=4GB \
+      --mem-type=LPDDR4_3200_2x16  \
+      --cpu-clock=2.5GHz \
+      --cpu-type=DerivO3CPU \
+      --ruby \
+      --access-backing-store \
+      --l2_size=2097152 \
+      --l2_assoc=16 \
+      --cacheline_size=32 \
+      --accel_cfg_file=gem5.cfg \
+      --fast-forward=10000000000 \
+      -c <path-to-smaug>/build/bin/smaug \
+      -o "my_model_topo.pbtxt my_model_params.pb --gem5 --debug-level=0"
+
+This command runs our custom 3-level model in gem5-Aladdin simulation.
```

Comment:
I think 3-level model refers to the DNN? If so, I think you meant three-layer DNN. Otherwise it can be confusing what 3 levels you are referring to (I initially thought it was L1/L2/L3 caches).

Reply:
Yes it's the DNN model. Clarified it.